src/disc-golf-events/DiscGolfEventDetails.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useState, useCallback } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { Container, Typography, Paper, CircularProgress, Alert, Button } from '@mui/material';
+import config from '../config';
+
+interface DiscGolfEvent {
+    id: string;
+    tournamentTitle: string;
+    tournamentDate: string;
+    region: string;
+    pdga?: string;
+    externalLink?: string;
+}
+
+const DiscGolfEventDetails = () => {
+    const { id } = useParams<{ id: string }>();
+    const navigate = useNavigate();
+    const [event, setEvent] = useState<DiscGolfEvent | null>(null);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<string>('');
+
+    const fetchEventDetails = useCallback(() => {
+        if (!id) {
+            setError('No tournament ID');
+            setLoading(false);
+            return;
+        }
+
+        setLoading(true);
+        setError('');
+
+        const url = `${config.discGolfServiceUrl}/public/events/${id}`;
+
+        fetch(url, {
+            method: 'GET',
+            credentials: 'include',
+            headers: {
+                'Content-Type': 'application/json'
+            }
+        })
+            .then(res => {
+                if (!res.ok) {
+                    throw new Error(`No tournament found (status: ${res.status})`);
+                }
+                return res.json();
+            })
+            .then((data: DiscGolfEvent) => {
+                setEvent(data);
+                setLoading(false);
+            })
+            .catch(err => {
+                console.error('Tournament download error:', err);
+                setError(err.message || 'The tournament data could not be downloaded.');
+                setLoading(false);
+            });
+    }, [id]);
+
+    useEffect(() => {
+        fetchEventDetails();
+    }, [fetchEventDetails]);
```

Comment:
Why dependency on whole fetchEventDetails? We should base on [id] in URL or even just empty array []